packages/core/src/agent/index.ts

```diff
@@ -1991,25 +2105,29 @@ Message ${msg.threadId && msg.threadId !== threadObject.id ? 'from previous conv
   async generate(
     messages: string | string[] | CoreMessage[] | AiMessageType[],
     args?: AgentGenerateOptions<undefined, undefined> & { output?: never; experimental_output?: never },
-  ): Promise<GenerateTextResult<any, undefined>>;
+  ): Promise<GenerateTextResultWithTripwire<any, undefined>>;
```

Comment:
all of these tripwire types shouldn't be a wrapping generic, we can just update the type and/or do `& TripwireProperties`